### PR TITLE
DIS-125 Update stored procedure to update a forum's thread count

### PR DESCRIPTION
**Pull request by jogura**
Update the forum's thread count as part of the post_stats update

@jimmordino @armonr

### CODE REVIEW

File: service/discussion/sql/create_discussion_db.sql [hunk 1]
```diff
@@ -451,11 +452,17 @@ CREATE PROCEDURE `update_post_stats` (IN siteId INTEGER, IN postId BIGINT)
     JOIN thread ON (post.thread_id = thread.id)
     WHERE thread.container_id = forumId AND thread.site_id = siteId;
 
-   IF postCount = -1 OR maxPostId = 0 THEN
+    SET threadCount = -1;
```
Comment (armonr) [inline service/discussion/sql/create_discussion_db.sql]: Isn't this initialized already?

File: service/discussion/sql/create_discussion_db.sql [hunk 2]
```diff
@@ -451,11 +452,17 @@ CREATE PROCEDURE `update_post_stats` (IN siteId INTEGER, IN postId BIGINT)
...
+    SELECT COUNT(thread.id) INTO threadCount
+    FROM forum
+    JOIN thread on (forum.id = thread.container_id)
```
Comment (armonr) [inline service/discussion/sql/create_discussion_db.sql]: `ON` capitalized :)

Comment (armonr) [inline service/discussion/sql/create_discussion_db.sql]: And join on container_type = 'forum'?

Comment (jimmordino) [inline service/discussion/sql/create_discussion_db.sql]: The second query is not necessary.  Just use
```
SELECT COUNT(post.id), MAX(post.id), COUNT(DISTINCT thread.id) INTO postCount, maxPostId, threadCount
```
in the first query.

Comment (jogura) [inline service/discussion/sql/create_discussion_db.sql]: That is way simpler thanks: 7c80c8f

**Comment by jimmordino** [comment]
:+1: , thanks

**Pull request merged**